websites/S/Stremio/presence.ts

```diff
@@ -0,0 +1,148 @@
+						)
+						?.firstElementChild?.getAttribute("src") ?? "logo";
+			}
+		} else if (page.includes("addons")) {
+			search = document.querySelector(
+				"#addons > div.filter > form > div.addon-search-input > input"
+			);
+			if (search && search.value) {
+				presenceData.details = "Searching Addons For:";
+				presenceData.state = search.value;
+			} else {
+				if (active) {
+					presenceData.state = `${active.charAt(0).toUpperCase()}${active.slice(
+						1,
+						active.length
+					)}`;
+				} else presenceData.state = "All";
+
+				presenceData.details = `Browsing ${
+					document.querySelectorAll("[class='ng-scope selected']")[0]
+						?.textContent
+				}:`;
+			}
+		} else if (page.includes("settings")) {
+			presenceData.details = `${
+				document.querySelectorAll("[class='ng-scope ng-binding active']")[0]
+					?.textContent ?? "General"
+			} Settings`;
+		} else if (page.includes("/discover/")) {
+			if (active) {
+				genreSort = document.querySelectorAll(
+					"[class='ng-scope ng-binding selected']"
+				)[0]?.textContent;
+				if (!genreSort) {
+					genreSort =
+						document.querySelectorAll(
+							"[class='ng-binding ng-scope selected']"
+						)[2]?.textContent ?? "None";
+				}
+				if (genreSort == "Top") genreSort = "All";
+				presenceData.details = `Browsing ${active}`;
+				presenceData.state = `Genre: ${genreSort}`;
+			} else presenceData.state = "Browsing Movies";
+		} else if (page.includes("/library")) {
+			genreSort = document
+				.querySelector<HTMLSelectElement>(
+					"#library > div.sort-filter > div.custom-select.lib-sort > select"
+				)
+				?.value.replace("SORT", "")
+				.replace("_", " ")
+				.trim();
+			if (genreSort) {
+				presenceData.state = `Sorted by: ${genreSort.substring(0, 1)}${genreSort
+					.substring(1, genreSort.length)
+					.toLowerCase()}`;
+			}
+			if (active) presenceData.details = `${active} Library`;
+			else presenceData.details = "Library";
+		} else if (page.includes("/calendar")) presenceData.details = "Calendar";
+		else if (page.includes("player")) {
+			if (video?.duration && video) {
+				timestamp = presence.getTimestampsfromMedia(video);
+				pauseCheck = video.paused;
+			} else if (!video.duration && document.querySelector("#controlbar-top")) {
+				const split = document
+					.querySelector("#play-progress-text")
+					?.textContent.split("/");
+				if (split && split[0]) {
+					timestamp = presence.getTimestamps(
+						presence.timestampFromFormat(split[0].trim()),
+						presence.timestampFromFormat(split[1].trim())
+					);
+				} else timestamp = void 0;
+
+				pauseCheck = document
+					.querySelector("#controlbar-top")
+					?.firstElementChild?.className.includes("pause")
+					? false
+					: true;
+			}
+			delete presenceData.startTimestamp;
+			if (
+				!pauseCheck &&
+				!document.querySelector("#loading-logo").className.includes("flashing")
+			) {
+				presenceData.endTimestamp = timestamp[1] ?? 0;
```

Comment:
I don't think it will be undefined?
```suggestion
				presenceData.endTimestamp = timestamp[1];
```